### ISSUE TITLE
Prevent function redeclaration / possible problem with no-func-assign

### ISSUE THREAD

**Issue by johnnyreilly** [eslint/eslint#3910]
Hi,

First of all, well done on eslint.  I gave it a try yesterday just to see if it was any better than jshint and once I got past the initial setup it started to find problems in my code that jshint hadn't picked up!  (Particularly around variable shadowing.)

Related to that I've been digging my through the rules and a rule to prevent function redeclaration doesn't seem to be in place.  ie Something that would prevent the following:

```js
function iDoNotWantMoreThanOneOfThese() {
}

function iDoNotWantMoreThanOneOfThese() {
}
```

I thought that [no-func-assign](http://eslint.org/docs/rules/no-func-assign) might do the job but actually it doesn't.  Is this something that isn't presently catered for?

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
Have you tried `no-redeclare`?

**Comment by johnnyreilly** [comment]
Bingo!  Thanks chap.
